scripts/chgres_cube.py

```diff
@@ -0,0 +1,204 @@
+    os.environ["MEMBER"] = member
+
+    # Find the names of the external model files
+    expt_config.dereference(
+        context={
+            "cycle": cycle,
+            **os.environ,
+            **expt_config,
+        }
+    )
+    chgres_cube_config = _walk_key_path(expt_config, key_path)
+    input_type = chgres_cube_config["chgres_cube"]["namelist"]["update_values"][
+        "config"
+    ].get("input_type")
+
+    varsfilepath = chgres_cube_config["input_files_metadata_path"]
+    external_config = get_yaml_config(varsfilepath)
+    external_config_fns = external_config["external_model_fns"]
+    external_config_fhrs = external_config["external_model_fhrs"]
+
+    if "task_make_ics" in key_path:
+        if input_type == "grib2":
+            os.environ["fn_grib2"] = external_config_fns[0]
+        else:
+            os.environ["fn_atm"] = external_config_fns[0]
+            os.environ["fn_sfc"] = external_config_fns[1]
+
+        chgres_cube_driver = ChgresCube(
+            config=config_file,
+            cycle=cycle,
+            key_path=key_path,
+            leadtime=dt.timedelta(hours=0),
+        )
+        rundir = Path(chgres_cube_driver.config["rundir"])
+        logging.info(f"Will run in {rundir}")
+        chgres_cube_driver.run()
+
+        if not (rundir / "runscript.chgres_cube.done").is_file():
+            logging.error("Error occurred running chgres_cube. See component error logs.")
+            sys.exit(1)
+
+        # Deliver output data to the forecast's INPUT dir
+        delivery_dir = rundir.parent / "INPUT"
+        _deliver_files(config=expt_config, dst_dir=delivery_dir, key_path=key_path, src_dir=rundir)
+
+    else: # loop over make_lbcs tasks
+        # This loop will need a version of the config that is not dereferenced
+        expt_config = get_yaml_config(config_file)
+
+        fhrs_and_fns = list(zip(external_config_fhrs, external_config_fns))
+        for external_fhr, external_fn in fhrs_and_fns:
+            if input_type == "grib2":
+                os.environ["fn_grib2"] = external_fn
+            else:
+                os.environ["fn_atm"] = external_fn
+
+            lbc_offset_fhrs = expt_config["task_get_extrn_lbcs"]["envvars"][
+                    "EXTRN_MDL_LBCS_OFFSET_HRS"
+                ]
+            fcst_hr_lam = int(external_fhr) - int(lbc_offset_fhrs)
+            leadtime = dt.timedelta(hours=fcst_hr_lam)
+
+            chgres_cube_driver = ChgresCube(
+                config=config_file,
+                cycle=cycle,
+                key_path=key_path,
+                leadtime=leadtime,
+            )
+            rundir = Path(chgres_cube_driver.config["rundir"])
+            logging.info(f"Will run in {rundir}")
+            chgres_cube_driver.run()
+
+            if not (rundir / "runscript.chgres_cube.done").is_file():
+                logging.error("Error occurred running chgres_cube. See component error logs.")
+                sys.exit(1)
```

Comment:
If I'm reading right, these lines are identical to those on (current line numbers) 126-138, except for the value of `leadtime=`. Can they be factored out to a function with the leadtime as a parameter?

Reply:
Hmm, now that the loop is largely factored out to the output files, I wonder - can we get away with reducing the ics/lbcs split to something like:
```
if "task_make_ics" in key_path:
    leadtime = dt.timedelta(hours=0)
    os.environ["fn_sfc"] = external_config_fns[1]
```
?